### ISSUE TITLE
Arc Browser

### ISSUE THREAD

**Issue by neoOpus** [aloneguid/bt#76]
Please add this New Browser to the detection list

[Arc](https://arc.net/)

**Comment by donaldguy** [comment]
Arc is weird, and seems to have only fully bound the notion of launching a url via the "CWA Protocol Association Execute Command" [as bound in a `DelegateExecute` registry key to the GUID (`{A56A841F-E974-45C1-8001-7E3F8A085917}`] ONLY WHEN ITS THE DEFAULT BROWSER

but after objectively more hours than it was worth, I discovered/verified that it will also end up working if one calls [`Windows.System.Launcher.LaunchAsyncUri`](https://learn.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-22621)

with `LauncherOptions` setting [`TargetApplicationPackageFamilyName`](https://learn.microsoft.com/en-us/uwp/api/windows.system.launcheroptions.targetapplicationpackagefamilyname) to `TheBrowserCompany.Arc_ttt1ap7aakyb4`

fundamentally this is all .NET stuff, I gather (its not my area of expertise), so it should be not hard to implement directly if one is so inclined, but as shared in https://github.com/U-C-S/Hurl/issues/100#issuecomment-2105612332 - I managed to force into PowerShell and then that PowerShell script inside a `.{bat,cmd}` container so it could be treated as an exe

I don't per se have a solution for the icon, but if you were inclined to do one of them "batch to exe" convertors, you could then probably just set the icon on that.

**Comment by aloneguid** [comment]
Arc won't install on Windows 11 at all. It does seems like a new and slightly unstable browser. It may be worth revisiting when it works properly?

![image](https://github.com/aloneguid/bt/assets/3155189/1fe55b48-aff2-42e9-8d05-43649dddec72)

**Comment by arendon1** [comment]
Commenting just to have this thread on my watchlist. I'm also interest in Arc support.

**Comment by aloneguid** [comment]
Still can't install it (I'm usually testing new browsers in Windows Sandbox). It seems Arc is only available for Windows 11?

**Comment by arendon1** [comment]
> Still can't install it (I'm usually testing new browsers in Windows Sandbox). It seems Arc is only available for Windows 11?

You're totally right,  current version of Arc are only Windows 11 exclusively, they seem to be working on Windows 10 version of it.

https://resources.arc.net/hc/en-us/articles/19400148671383-Unsupported-Devices

Is this a major issue to recognize it as a browser?

**Comment by aloneguid** [comment]
@arendon1 Honestly I don't know because I can't even install it on Widnows 11 (see error above).

**Comment by neoOpus** [comment]
> @arendon1 Honestly I don't know because I can't even install it on Widnows 11 (see error above).

Actually, I don't have any problem installing it, if you want, I can provide you a remote control to a fresh Windows 11 today or this weekend so you can do your testing and updates… Please let me know if you are interested and what remote control software do you want to use (RustDesk, AnyDesk, Teamviewer, Parsec)

**Comment by link6155** [comment]
I have not tried this myself, but there's a workaround for a similar tool here https://github.com/U-C-S/Hurl/issues/100#issuecomment-2113632225

**Comment by aloneguid** [comment]
@neoOpus I appreciate the offer, but it seems Arc haven't figured a way to create a way to install it. I've seen reddit comments people suggesting modifying their installer, installing custom certificates, setting up Windows Store correctly, enabling developer mode and so on which seems like a terrible way to support this. Until Arc supports all current Windows versions and not a specific edition of Windows 11 with custom configuration it's a no go for me.
The other way to support this is to write a custom program that opens Arc and then invoke it from BT, but that might be a waste of energy because Arc people may create a proper installer by then.

**Comment by Theblockbuster1** [comment]
I was thinking it might be possible to open a URL via a browser extension. Also the browser would have to be opened if it's not running.

For now I find it easier to just have my default browser as Arc, and use the BT extension to open links in Firefox in 1 click.

**Comment by neoOpus** [comment]
[**@Theblockbuster1**](https://github.com/Theblockbuster1) commented on [Jun 28, 2024, 4:01 PM EDT](https://github.com/aloneguid/bt/issues/76#issuecomment-2197549966 "2024-06-28T20:01:54Z - Replied by Github Reply Comments"):
> I was thinking it might be possible to open a URL via a browser extension. Also the browser would have to be opened if it's not running.
> 
> For now I find it easier to just have my default browser as Arc, and use the BT extension to open links in Firefox in 1 click.
> 
> _Originally posted by @Theblockbuster1 in https://github.com/aloneguid/bt/issues/76#issuecomment-2197549966_

There is a solution for that through the BT chrome extension available in the Chrome store or linked on the webpage created by Ivan @aloneguid. This extension enables you to right-click on the link and select the option from the context menu to open it in BT (app), which will then suggest the browsers you have installed to choose from.

**Comment by Theblockbuster1** [comment]
Yes, that's what I'm using :) it's very useful. I mostly use it by clicking on the extension icon to reopen the current tab.

**Comment by neoOpus** [comment]
There are other extensions that can do the same thing like these

https://webextension.org/listing/open-in.html

You pick the one you just need but you can opt to configure General Purpose

**Comment by aloneguid** [comment]
Seems like Arc will be coming to windows 10, and I think this might make it a normal recognisable browser for all windows versions. See https://browserco.typeform.com/to/k37wtsev?typeform-source=arc.net#source=download

**Comment by aloneguid** [comment]
@donaldguy I have solved the issue of opening links in Arc via UWP API in Windows, so it's coming along. What's left is detecting whether Arc is installed.

**Comment by aloneguid** [comment]
At the moment I'm able to detect Arc browser (and Firefox Windows Store version as a bonus)

![image](https://github.com/aloneguid/bt/assets/3155189/d0b07366-df2a-4e39-b36a-69e76faca99e)

The only outstanding issue is finding out where Arc stores chromium profiles on disk (looking for help here). Arc itself needs to be launched differently via UWP COM API but i've done this already.

**Comment by neoOpus** [comment]
> At the moment I'm able to detect Arc browser (and Firefox Windows Store version as a bonus)
> 
> ![image](https://private-user-images.githubusercontent.com/3155189/346982045-d0b07366-df2a-4e39-b36a-69e76faca99e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1NzgzNjgsIm5iZiI6MTcyMDU3ODA2OCwicGF0aCI6Ii8zMTU1MTg5LzM0Njk4MjA0NS1kMGIwNzM2Ni1kZjJhLTRlMzktYjM2YS02OWU3NmZhY2E5OWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTBUMDIyMTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWFhZTA0MDYzZjBhN2JhNTNjM2VjOTg5YjdmNzEwZmVkMzBkZjA3MmQ4MmE4YzU5MTAyMzVmMGNjM2Y1MmQ1MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.WkCLrryO1qNowGuFU1xPCXqmDfA9_wQT3QratPjhfE8)
> 
> The only outstanding issue is finding out where Arc stores chromium profiles on disk (looking for help here). Arc itself needs to be launched differently via UWP COM API but i've done this already.


Unfortunately, I do not have access to a Windows machine, so I won't be able to investigate anything personally. Additionally, the profile is likely behind a random naming scheme for security reasons. However, the Arc team may be able to provide a way for you to access the information you need. Do you need me to contact Arc on your behalf?

**Comment by aloneguid** [comment]
No worries @neoOpus thanks, I found it, and it's under `..\AppData\Local\Packages\TheBrowserCompany.Arc_ttt1ap7aakyb4\LocalCache\Local\Arc\User Data\Local State` which is a normal Chromium data folder with all the usual stuff inside.

What I've got left is passing an argument to Arc which it seems to ignore at the moment. It's not UWP issue as I can do it now for Firefox MSSTORE version.

**Comment by neoOpus** [comment]
> No worries @neoOpus thanks, I found it, and it's under `..\AppData\Local\Packages\TheBrowserCompany.Arc_ttt1ap7aakyb4\LocalCache\Local\Arc\User Data\Local State` which is a normal Chromium data folder with all the usual stuff inside.
> 
> What I've got left is passing an argument to Arc which it seems to ignore at the moment. It's not UWP issue as I can do it now for Firefox MSSTORE version.

I think this part of the path "ttt1ap7aakyb4" is random and changes from one user to another which could be a problem but if the field can be edited then anyone can use the appropriate path by looking in their folders tree

**Comment by aloneguid** [comment]
> Arc is weird, and seems to have only fully bound the notion of launching a url via the "CWA Protocol Association Execute Command" [as bound in a `DelegateExecute` registry key to the GUID (`{A56A841F-E974-45C1-8001-7E3F8A085917}` ONLY WHEN ITS THE DEFAULT BROWSER]
> 
> but after objectively more hours than it was worth, I discovered/verified that it will also end up working if one calls [`Windows.System.Launcher.LaunchAsyncUri`](https://learn.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-22621)
> 
> with `LauncherOptions` setting [`TargetApplicationPackageFamilyName`](https://learn.microsoft.com/en-us/uwp/api/windows.system.launcheroptions.targetapplicationpackagefamilyname) to `TheBrowserCompany.Arc_ttt1ap7aakyb4`
> 
> fundamentally this is all .NET stuff, I gather (its not my area of expertise), so it should be not hard to implement directly if one is so inclined, but as shared in [U-C-S/Hurl#100 (comment)](https://github.com/U-C-S/Hurl/issues/100#issuecomment-2105612332) - I managed to force into PowerShell and then that PowerShell script inside a `.{bat,cmd}` container so it could be treated as an exe
> 
> I don't per se have a solution for the icon, but if you were inclined to do one of them "batch to exe" convertors, you could then probably just set the icon on that.

This is really interesting. Arc does not open URL even if it's the default browser for me.

**Comment by aloneguid** [comment]
It seems to work now

![image](https://github.com/user-attachments/assets/e8286c8d-0033-48fa-9faa-2258492adfc4)

**Comment by aloneguid** [comment]
It's now released in 4.0.1, please try.

**Comment by arendon1** [comment]
These are amazing news! tysm for your hard work Ivan!

**Comment by arendon1** [comment]
Can confirm prev commenter case. After Arc's update had to refresh the browser discovery on bt.
Not a big issue, but a behavior worth noting, I think.

**Comment by aloneguid** [comment]
I can confirm this as well, rediscover does fix it, but that's not ideal. Issue is that on update msstore apps will change the folder name where they are installed, so I need to handle this as well, should be fixed soon.

**Comment by neoOpus** [comment]
> > I can confirm this as well, rediscover does fix it, but that's not ideal. Issue is that on update msstore apps will change the folder name where they are installed, so I need to handle this as well, should be fixed soon. The good news is browser ID is not changed, so all the confiruration is kept on rediscover.
> 
> Would caching the browser Icon be a possible solution?

Maybe the discover should be reworked... if it finds the browser in the setting file then it should just update the path and keep it at the same position without messing with other settings unless necessary... but it is up to Ivan!

**Comment by killianconsulting** [comment]
New issue with Arc Browser. It has been working but the actual program keeps crashing. If Browser Tamer is opened, I can see my browsers. In this case, Google Chrome Dev, Microsoft Edge, and Arc. The logo for Arc is the generic browser logo and if I run the rediscover browsers it'll either add Arc with the correct logo but then if you click on it to setup the profile it will crash or it'll just crash after attempting to rediscover browsers.
